core/templates/dev/head/services/UserService.js

```diff
@@ -66,6 +66,23 @@ oppia.factory('UserService', [
           data: newProfileImageDataUrl
         });
       },
+      getLoginAndLogoutUrls: function() {
+        var urlParameters = {
+          current_url: $window.location.href
+        };
+        var loginGenerationConfig = {
+          params: urlParameters
+        };
+        return $http.get('/url_handler', loginGenerationConfig).then(
+          function(response) {
+            if (response.data !== null) {
```

Comment:
I think that this is not needed because data can't be empty since that would mean that the request failed and this function wouldn't really be executed.

Reply:
Oh Okay, I actually put it there as a precaution but since it is not needed I have removed it now.